src/Aspire.Hosting.AWS/CloudFormation/ICloudFormationResource.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Amazon.CloudFormation;
+using Amazon.CloudFormation.Model;
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.AWS.CloudFormation;
+
+/// <summary>
+/// Resource representing an AWS CloudFormation stack.
+/// </summary>
+public interface ICloudFormationResource : IResource
+{
+    /// <summary>
+    /// Configuration for creating service clients from the AWS .NET SDK.
+    /// </summary>
+    IAWSSDKConfig? AWSSDKConfig { get; set; }
+
+    /// <summary>
+    /// The configured Amazon CloudFormation service client used to make service calls. If this property set
+    /// then AWSSDKConfig is ignored.
+    /// </summary>
+    IAmazonCloudFormation? CloudFormationClient { get; set; }
+
+    /// <summary>
+    /// The output parameters of the CloudFormation stack.
+    /// </summary>
+    List<Output>? Outputs { get; }
```

Comment:
Should this be a dictionary instead?

Reply:
In an ideal world yes but CloudFormation's API is modeled as returning a list of `Output` so the AWS .NET SDK is generated this way to return a `List<Output>`. This is exposing the data as it comes from the SDK as opposed to transforming it.

Reply:
You can transform it if that's the most usable API. This doesn't need to reflect it 1:1 but that's up to you.

Reply:
I would rather keep to mirror the SDK. Users also aren't likely to directly interact with this property. It is used between the provisioner and the `WithEnvironment` callback.